### DISCUSSION TITLE
Is there a way to use it offline on the same router.

### DISCUSSION THREAD

**Discussion by Enderteck** [schlagmichdoch/PairDrop#149] [Q&A]
Would I need to download the PWA on all devices ? Will they then see each other through LAN ? That would be amazing thx.

**Comment by Enderteck** [comment]
Oh and I could also pair all the devices in advance if that helps

**Comment by schlagmichdoch** [comment]
As of now PairDrop relies on a signal server + stun/turn server that must be available. In theory you could host an instance of PairDrop on your local network including a locally available STUN server. This should be sufficient to make devices connect to each other.

The focus of PairDrop has always been that no setups needed for devices to connect to each other everything runs on a Webbrowser.

In the future I might create an app version that lets you host a PairDrop instance on your phone, so even when offline, all devices connecting to your IP address via hotspot or local WiFi can use PairDrop.

If you want to connect your own devices and are willing to install software for that, I strongly suggest LocalSend. I’m not the biggest fan of the UI but the functionality is great: 
https://github.com/localsend/localsend

**Comment by Enderteck** [comment]
> On the future I might create an app version that lets you host a PairDrop instance on your phone, so even when offline, all devices connecting to your IP address via hotspot or local WiFi can use PairDrop.

That would be amazing, thanks for the answer
